### ISSUE TITLE
Ollama models not working

### ISSUE THREAD

**Issue by murillocamargo** [mckaywrigley/chatbot-ui#1084]
Ollama models are being loaded correctly in the UI, but while trying to chat says models can't be found.

![image](https://github.com/mckaywrigley/chatbot-ui/assets/7115912/eb5c0ebd-dc5b-4ec7-ae31-f04e52aef0e1)

![image](https://github.com/mckaywrigley/chatbot-ui/assets/7115912/206b23cb-35e0-4d3f-a4b7-e13db8a6cf30)

![image](https://github.com/mckaywrigley/chatbot-ui/assets/7115912/78ac1ba1-b5d7-4152-b0f2-f524b977ce97)

**Comment by mckaywrigley** [comment]
Try it now with the new PR. I suspect it will help.

**Comment by murillocamargo** [comment]
Hello, thanks for your reply and work so far.
I'm preparing a pull request for image clipboard paste support, but this is getting in my way.

Sorry, but the latest pull broke the application for me.

Here the details:

After running ´npm run chat´ I have 2 errors on the initial screen:

![image](https://github.com/mckaywrigley/chatbot-ui/assets/7115912/c50e3d11-3624-4745-a41b-59e4abc57518)

![image](https://github.com/mckaywrigley/chatbot-ui/assets/7115912/b0422fe0-462c-4b9b-90d3-15de92844527)

**Full terminal output:**

npm run chat

> chatbot-ui@2.0.0 chat
> supabase start && npm run db-types && npm run dev

supabase start is already running.
Run supabase status to show status of local Supabase containers.

> chatbot-ui@2.0.0 db-types
> supabase gen types typescript --local > supabase/types.ts

(node:1) ExperimentalWarning: Import assertions are not a stable feature of the JavaScript language. Avoid relying on their current behavior and syntax as those might change in a future version of Node.js.
(Use `node --trace-warnings ...` to show where the warning was created)
(node:1) ExperimentalWarning: Importing JSON modules is an experimental feature and might change at any time

> chatbot-ui@2.0.0 dev
> next dev

   ▲ Next.js 14.0.2
   - Local:        http://localhost:3000
   - Environments: .env.local

 ✓ Ready in 1706ms
 ✓ Compiled /middleware in 199ms (124 modules)
 ○ Compiling / ...
 ✓ Compiled / in 2.7s (1217 modules)
 ✓ Compiled in 594ms (714 modules)
 ✓ Compiled /api/localhost/ollama in 270ms (507 modules)
 ⨯ TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11457:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: Error: connect ECONNREFUSED 127.0.0.1:11434
      at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    errno: -111,
    code: 'ECONNREFUSED',
    syscall: 'connect',
    address: '127.0.0.1',
    port: 11434
  }
}
 ○ Compiling /chat ...
 ✓ Compiled /chat in 13.6s (3612 modules)

----

The UI loads correctly, but I can't chat and models list not being loaded.

![image](https://github.com/mckaywrigley/chatbot-ui/assets/7115912/a347e1be-7591-4c2c-8c5c-84de9d65a226)

---

**UPDATE**

After clearing my browser cache I was able to pass initial screen but got blocked on´/setup´.

Here the error, still something related to Ollama.

![image](https://github.com/mckaywrigley/chatbot-ui/assets/7115912/71597546-ddb8-4721-83e4-96a581188b1e)

**Comment by wrapss** [comment]
> Hello, thanks for your reply and work so far. I'm preparing a pull request for image clipboard paste support, but this is getting 

@murillocamargo  your ollama server doesn't seem to be responding? 
What is the response when you open http://127.0.0.1:11434?

**Comment by murillocamargo** [comment]
Thanks @wrapss 

Didn't realized had to start manually Ollama with `ollama serve` after latest updates.

Before pulling latest code, the application worked without initiating Ollama. 
However, when I attempted to start Ollama, I received a notification that it was already active following the `npm run chat` command.

Problem solved.

**Comment by murillocamargo** [comment]
Hello @wrapss but I'm still getting this error.

The application is working but Ollama models still are not being found.

My Ollama terminal output:
![image](https://github.com/mckaywrigley/chatbot-ui/assets/7115912/64b5518a-99b4-44ec-8015-71717c99e68e)

![image](https://github.com/mckaywrigley/chatbot-ui/assets/7115912/382375de-fcf0-45c8-aff8-2e7574143166)

**Comment by murillocamargo** [comment]
Hello @wrapss but I'm still getting this error.

The application is working, Ollama models are being listed but I get this error trying to chat.

My Ollama terminal output:
![image](https://github.com/mckaywrigley/chatbot-ui/assets/7115912/64b5518a-99b4-44ec-8015-71717c99e68e)

![image](https://github.com/mckaywrigley/chatbot-ui/assets/7115912/382375de-fcf0-45c8-aff8-2e7574143166)

**Comment by wrapss** [comment]
if you can modify the code, go to components/chat/chat-helpers/index.ts and add a console.log(chatSettings.model) around line 157 (above fetchChatResponse), once the modification is done try a new chat and check the model name in the console.

**Comment by murillocamargo** [comment]
Console log, it appears it's logging the model name correctly:

![image](https://github.com/mckaywrigley/chatbot-ui/assets/7115912/d9734535-40c8-4ca7-a925-82cc74ad0573)

**Comment by wrapss** [comment]
> [murillocamargo](/murillocamargo)

it's weird because it seems to be a problem that comes from ollama, you can look in the network tab what gives the payload and the response?

**Comment by murillocamargo** [comment]
Thanks again @wrapss 

Here the payload and the response was 404.

![image](https://github.com/mckaywrigley/chatbot-ui/assets/7115912/e424b66f-427c-4ed5-b197-d841eb0be2ef)

![image](https://github.com/mckaywrigley/chatbot-ui/assets/7115912/4186c99e-0ab3-4c5a-817f-402bbb21e632)

![image](https://github.com/mckaywrigley/chatbot-ui/assets/7115912/79b1d30e-3e01-434d-bf6b-e42b7fe82314)

**Comment by wrapss** [comment]
> Here the payload and the response was 404.
> 


strange, so the problem comes from ollama, what version do you have? I know they added the chat api recently so maybe try updating ollama.

**Comment by murillocamargo** [comment]
> > Here the payload and the response was 404.
> 
> strange, so the problem comes from ollama, what version do you have? I know they added the chat api recently so maybe try updating ollama.

Will try updating Ollama, thanks

**Comment by wrapss** [comment]
@murillocamargo [update](https://github.com/jmorganca/ollama/pull/1392) dated december 5

**Comment by murillocamargo** [comment]
Hello @wrapss 

Just reporting back, updating Ollama solved the issue, I can now chat with their models.

Thanks for your help.

Closing this now.
